haskell-mode.el

```diff
@@ -587,7 +587,8 @@ executable found in PATH.")
                 (put-text-property (match-beginning 3) (1+ (match-end 3)) 'syntax-table (string-to-syntax "|")))))
            ((equal token-kind 'template-haskell-quasi-quote)
             (put-text-property (match-beginning 2) (match-end 2) 'syntax-table (string-to-syntax "\""))
-            (put-text-property (match-beginning 4) (match-end 4) 'syntax-table (string-to-syntax "\""))))
+            (put-text-property (match-beginning 4) (match-end 4) 'syntax-table (string-to-syntax "\""))
+            (put-text-property (match-beginning 3) (match-end 3) 'syntax-table (string-to-syntax "."))))
```

Comment:
This is a cannon used on a fly and will break some other functionality, like navigation word-by-word inside quasiquote.

Better: loop using re-search-forward to find all `"` and put `.` syntax only on those.